nsq/reader.py

```diff
@@ -677,7 +669,7 @@ def _redistribute_rdy_state(self):
             # We also don't attempt to avoid the connections who previously might have had RDY 1
             # because it would be overly complicated and not actually worth it (ie. given enough
             # redistribution rounds it doesn't matter).
-            possible_conns = list(self.conns.values())
+            possible_conns = [c for c in conns if not (c.in_flight or c.rdy)]
```

Comment:
With the proposed changes, the full diff would look like https://github.com/nsqio/pynsq/compare/master...alpaker:no-decr-total-rdy-2?diff=unified&expand=1&name=no-decr-total-rdy-2#diff-2a8bc85bf9c95f482da1eb0490a60251R654